crates/dbsp/src/algebra/decimal/precision.rs

```diff
@@ -0,0 +1,103 @@
+use rust_decimal::Decimal;
+use size_of::SizeOf;
+use std::fmt::Display;
+
+// Should be kept in sync with the Java code in DBSPTypeDecimal.java
+const MIN_DECIMAL_PRECISION: u32 = 1;
+const MAX_DECIMAL_PRECISION: u32 = 38;
+
+#[derive(
+    Debug,
+    Clone,
+    Copy,
+    PartialEq,
+    Eq,
+    PartialOrd,
+    Ord,
+    Hash,
+    SizeOf,
+    rkyv::Archive,
+    rkyv::Serialize,
+    rkyv::Deserialize,
+)]
+#[cfg_attr(feature = "with-serde", derive(serde::Serialize, serde::Deserialize))]
+pub struct Precision(u32);
+
+impl Precision {
+    pub fn extract_precision_from_decimal(val: Decimal) -> Result<Self, String> {
```

Comment:
do we need this?
does ilog work if the mantissa is negative?
why is it the max of the scale and the log?